rpc/websockets.go

```diff
@@ -359,7 +359,7 @@ type pubSubAPI struct {
 }
 
 // newPubSubAPI creates an instance of the ethereum PubSub API.
-func newPubSubAPI(clientCtx client.Context, logger log.Logger, tmWSClient *rpcclient.WSClient) *pubSubAPI {
+func newPubSubAPI(clientCtx client.Context, logger log.Logger, tmWSClient rpcclient.EventsClient) *pubSubAPI {
 	logger = logger.With("module", "websocket-client")
 	return &pubSubAPI{
 		events:    rpcfilters.NewEventSystem(logger, tmWSClient),
```

Comment:
can we replace tmWSClient with clientCtx.Client to avoid passing in api creator

Reply:
we can, just to keep it incremental, there are much more refactoring to come